packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m

```diff
@@ -250,15 +250,15 @@ - (void)onMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
     result(nil);
   } else if ([call.method isEqualToString:@"map#takeSnapshot"]) {
     if (self.mapView != nil) {
-      UIGraphicsImageRendererFormat *format = [UIGraphicsImageRendererFormat defaultFormat];
-      format.scale = [[UIScreen mainScreen] scale];
       UIGraphicsImageRenderer *renderer =
-          [[UIGraphicsImageRenderer alloc] initWithSize:self.mapView.frame.size format:format];
-
+          [[UIGraphicsImageRenderer alloc] initWithSize:self.mapView.bounds.size];
+      // For some unknown reason mapView.layer::renderInContext API returns a blank image on iOS 17. So we have to use drawViewHierarchyInRect API.
       UIImage *image = [renderer imageWithActions:^(UIGraphicsImageRendererContext *context) {
-        [self.mapView.layer renderInContext:context.CGContext];
+        [self.mapView drawViewHierarchyInRect:self.mapView.bounds afterScreenUpdates:YES];
       }];
-      result([FlutterStandardTypedData typedDataWithBytes:UIImagePNGRepresentation(image)]);
+      NSData *imageData = UIImagePNGRepresentation(image);
+      FlutterStandardTypedData *typedData = [FlutterStandardTypedData typedDataWithBytes:imageData];
```

Comment:
This can be inlined into the `result` call; having the separate variable just duplicates the type, which makes this less readable IMO.

Reply:
done.